cmd/edit/machinepool/cmd_test.go

```diff
@@ -1,33 +1,176 @@
+				// First get
+				t.ApiServer.AppendHandlers(RespondWithJSON(http.StatusOK, nodePoolResponse))
+				// Edit
+				t.ApiServer.AppendHandlers(RespondWithJSON(http.StatusOK, ""))
+				t.SetCluster(clusterId, mockClusterReady)
+				args := NewEditMachinepoolUserOptions()
+				args.machinepool = nodePoolId
+				runner := EditMachinePoolRunner(args)
+				cmd := NewEditMachinePoolCommand()
+				Expect(cmd.Flags().Set("enable-autoscaling", "true")).To(Succeed())
+				Expect(cmd.Flags().Set("labels", "test=test")).To(Succeed())
+				Expect(cmd.Flag("cluster").Value.Set(clusterId)).To(Succeed())
+				Expect(cmd.Flags().Set("min-replicas", "2")).To(Succeed())
+				Expect(cmd.Flags().Set("max-replicas", "10")).To(Succeed())
+				Expect(cmd.Flag("yes").Value.Set("true")).To(Succeed())
+				Expect(runner(context.Background(), t.RosaRuntime, cmd,
+					[]string{"--machinepool", nodePoolId})).To(Succeed())
+			})
+			It("No need for --machinepool (ID by itself)", func() {
+				t.ApiServer.AppendHandlers(RespondWithJSON(http.StatusOK, hypershiftClusterReady))
+				// First get
+				t.ApiServer.AppendHandlers(RespondWithJSON(http.StatusOK, nodePoolAutoResponse))
+				// Edit
+				t.ApiServer.AppendHandlers(RespondWithJSON(http.StatusOK, ""))
+				t.SetCluster(clusterId, mockClusterReady)
+				args := NewEditMachinepoolUserOptions()
+				args.machinepool = nodePoolId
+				args.autoscalingEnabled = true
+				runner := EditMachinePoolRunner(args)
+				cmd := NewEditMachinePoolCommand()
+				Expect(cmd.Flag("cluster").Value.Set(clusterId)).To(Succeed())
+				Expect(cmd.Flag("yes").Value.Set("true")).To(Succeed())
+				Expect(cmd.Flags().Set("labels", "test=test")).To(Succeed())
+				Expect(cmd.Flags().Set("enable-autoscaling", "true")).To(Succeed())
+				Expect(cmd.Flags().Set("min-replicas", "2")).To(Succeed())
+				Expect(cmd.Flags().Set("max-replicas", "10")).To(Succeed())
+				Expect(runner(context.Background(), t.RosaRuntime, cmd, []string{nodePoolId, "--min-replicas",
+					"2", "--enable-autoscaling", "true", "--interactive", "false", "--max-replicas",
+					"10"})).To(Succeed())
+			})
+		})
+	})
 })
+
+// formatNodePool simulates the output of APIs for a fake node pool list
+func formatNodePool() string {
+	version := cmv1.NewVersion().ID("4.12.24").RawID("openshift-4.12.24")
+	awsNodePool := cmv1.NewAWSNodePool().InstanceType("m5.xlarge")
+	nodeDrain := cmv1.NewValue().Value(1).Unit("minute")
+	nodePool, err := cmv1.NewNodePool().ID(nodePoolId).Version(version).
+		AWSNodePool(awsNodePool).AvailabilityZone("us-east-1a").NodeDrainGracePeriod(nodeDrain).Build()
+	Expect(err).ToNot(HaveOccurred())
+	return fmt.Sprintf("{\n  \"items\": [\n    %s\n  ],\n  \"page\": 0,\n  \"size\": 1,\n  \"total\": 1\n}",
+		test.FormatResource(nodePool))
+}
+
+// formatNodePoolAutoscaling simulates the output of APIs for a fake node pool list with autoscaling
+func formatNodePoolAutoscaling() string {
```

Comment:
@hunterkepley Can we move these into `test` package alongside the other "FormatXXX` functions that we define there. It would also be good to see if there is anyway to make them re-use any of the existing functions